keylime/ima/ima_dm.py

```
@@ -189,12 +189,13 @@ def validate_device_resume(self, event: "ResumeEvent", match_key) -> Failure:
             return failure
 
         # Check if the table hash is consistent
+        assert device_state.active_table_hash is not None
```

Would defer.

---

What about this? https://github.com/keylime/keylime/pull/1180/commits/0b02ca3b4dbd6ebdefd0f5ef4dd7da7d79a39aac

---

I think it is an internal error, although my understanding is we want to avoid throwing exceptions from the verifier.

---

If I remember correctly that the active_table_hash is None should never happen, so if it does it is an internal error.

>  I think it is an internal error, although my understanding is we want to avoid throwing exceptions from the verifier.

Yes correct, but this masks an potential error. The current solution is fine for now, but open also an issue so that we can investigate this further.

---

I propose a more radical approach: https://github.com/keylime/keylime/commit/9f1d01eda80b71095d685437e63335028a44fe32

---

Since @stefanberger sounds happy about the radical approach, I will call it resolved unless @THS-on has any counterargument.

---

Looking at the code it seems that the active_table_hash currently never none, so the more radical approach should be fine.